rholang/src/test/scala/coop/rchain/rholang/interpreter/accounting/RholangMethodsCostsSpec.scala

```diff
@@ -0,0 +1,688 @@
+        }
+      }
+    }
+  }
+
+  "keys" when {
+    "called on Map" should {
+      "charge constant cost" in {
+        implicit val errorLog = new ErrorLog()
+        val maps = Table[Map[Par, Par]](
+          "map",
+          emptyMap,
+          mapN(1, GInt(1)),
+          mapN(100, GInt(1)),
+          map(Seq((toRholangMap(mapN(10, GInt(10))), GInt(1))))
+        )
+        forAll(maps) { map =>
+          test(methodCall("keys", toRholangMap(map), List()), KEYS_METHOD_COST)
+        }
+      }
+    }
+  }
+
+  "size" when {
+    "called on Map" should {
+      "charge proportionally to the number of elements in the base Map" in {
+        implicit val errorLog = new ErrorLog()
+        val maps = Table[Map[Par, Par]](
+          "map",
+          emptyMap,
+          mapN(1, GInt(1)),
+          mapN(100, GInt(1)),
+          map(Seq((toRholangMap(mapN(10, GInt(10))), GInt(1))))
+        )
+        forAll(maps) { map =>
+          test(methodCall("size", toRholangMap(map), List()), sizeMethodCost(map.size))
+        }
+      }
+    }
+
+    "called on Set" should {
+      "charge proportionally to the number of elements in the base Set" in {
+        implicit val errorLog = new ErrorLog()
+        val sets = Table[Set[Par]](
+          "set",
+          emptySet,
+          Set(GInt(1)),
+          setN(100),
+          Set(toRholangMap(mapN(10, GInt(10))))
+        )
+        forAll(sets) { set =>
+          test(methodCall("size", toRholangSet(set), List()), sizeMethodCost(set.size))
+        }
+      }
+    }
+  }
+
+  "length" when {
+    "called on List" should {
+      "charge constant cost" in {
+        implicit val errorLog = new ErrorLog()
+        val lists = Table[Vector[Par]](
+          "list",
+          Vector.empty[Par],
+          Vector[Par](GInt(1)),
+          setN(100).toVector,
+          Vector[Par](toRholangMap(mapN(10, GInt(10))))
+        )
+        forAll(lists) { vector =>
+          test(methodCall("length", toRholangList(vector), List()), LENGTH_METHOD_COST)
+        }
+      }
+    }
+
+    "called on String" should {
+      "charge constant cost" in {
+        implicit val errorLog = new ErrorLog()
+        val strings = Table[GString](
+          "string",
+          GString(""),
+          GString("abcd"),
+          GString(Seq.fill(10000)("").mkString)
+        )
+        forAll(strings) { string =>
+          test(methodCall("length", Par(exprs = Seq(string)), List()), OP_CALL_COST)
+        }
+      }
+    }
+  }
+
+  "slice" when {
+    "called on List" should {
+      "charge proportionally to the number of elements it has to traverse" in {
+        implicit val errorLog = new ErrorLog()
+        val lists = Table[Vector[Par], Int, Int](
+          ("list", "from", "to"),
+          (Vector.empty[Par], 0, 1),
+          (Vector.empty[Par], 1, 0),
+          (Vector[Par](GInt(1)), 0, 1),
+          (setN(100).toVector, 10, 20),
+          (Vector[Par](toRholangMap(mapN(10, GInt(10)))), 0, 1)
```

Comment:
I've checked that all the collections/structures for which we provide index-like access first check if the requested index is within the size of the collection and if it's not it throws. It means that we never do any action. This won't be _free_ though because we also charge for the action of invoking rholang method.